### ISSUE TITLE
NVTabular Docker is Out of Date [BUG]

### ISSUE THREAD

**Issue by darien-schettler** [NVIDIA/NVTabular#548]
**Describe the bug**
The newest NVTabular docker on NGC (nvcr.io/nvidia/nvtabular:0.3) is out of date (at least CuDF and NVTabular are) by at least a few versions. Enough that it doesn't work with the tutorials/documentation found online. It is also made more difficult because **versioneer** is broken. This makes it so you can't see what version you have (either with PIP or through other means).

**Steps/Code to reproduce bug**
1. Download the newest docker
2. Try and follow along with the tutorials on the Github/Documentation (not what's included with the Docker)
3. Find errors due to old versioning

**Expected behavior**
I expect the docker to be updated regularly when changes are pushed to Github (especially breaking ones)

**Environment details (please complete the following information):**
 - Environment Location: Docker on GCP VM
 - Method of NVTabular install: Docker
   - docker pull nvcr.io/nvidia/nvtabular:0.3

**Comment by benfred** [comment]
Thanks for the bug report!

The versioneer tags on the container are wrong, and we've fixed so that they will be set appropriately for the next release.  This was a problem with our CI process for creating the container - it didn't pull in the git tags from the repo (just the source)  - which messed up versioneer. 

The container should have both NVTabular 0.3 and cudf 0.16 installed on it though (despite what the ```__version__``` strings indicate). This corresponds to the last release of NVTabular we did (0.3).

Our documentation is versioned - are you following along with the 0.3 documentation https://nvidia.github.io/NVTabular/v0.3.0/Introduction.html or the docs for the main branch ?  If so can you point out what errors you are seeing?

The next version of nvtabular (0.4) is scheduled for release on Feb 22nd - and the current unstable version of this is up on the 'main' branch. This version contains some breaking api changes, so if you try to run the notebooks/examples from the main branch on the 0.3 container you will get all sorts of different errors.

If you want to run the main branch, the easiest way right now is to use the 0.3 container and then go ```pip install --upgrade git+https://github.com/NVIDIA/NVTabular.git``` to grab the latest version of NVT from source.  If you hit any problems with this - please let me know!

**Comment by darien-schettler** [comment]
Thanks for the response **benfred**.

Is there any reason you wouldn't want to update the NVTabular docker to contain the most recent release of CuDF? 
I'm using NVTabular as part of the Merlin workflow so having up-to-date utilities in that workflow is pivotal. 

It seems counterintuitive that the CuDF version is 2 versions behind the currently released one.
  - Is there something that breaks when the CuDF version is updated to the current version? If not then why wouldn't you want to update the NGC docker accordingly? 

And no I did not know the documentation is versioned. I went to the main documentation page called out from Google. It is the newest as far as I can see.  However it does this...

```
cat_features = CATEGORICAL_COLUMNS >> Categorify(freq_threshold=15, out_path=OUTPUT_DATA_DIR)
cont_features = CONTINUOUS_COLUMNS >> FillMissing() >> Clip(min_value=0) >> LogOp() >> Normalize()
features = cat_features + cont_features + LABEL_COLUMNS
```

... which does not work in the provided version of NVTabular in the most recent docker.

---

***Is it not a problem that your most up-to-date docker (the one you recommend on your Github) does not work with your most up-to-date documentation? Please at least call this out somewhere on the documentation (i.e. In bold-red-text, this documentation does not work with the NGC NVTabular docker, please click here for the older versioned documentation that will work with that particular docker image)***

---
 
For now, I will continue to patch this by building my own docker off the base rapids docker and adding in the NVTabular and HugeCTR functionality separately.

**Comment by darien-schettler** [comment]
Also, on your main README it tells you how to launch the docker and then provides this link to go to the documentation to see examples.

**https://nvidia.github.io/NVTabular/main/examples/index.html**

This clearly isn't the v0.3 version of the documentation. So the current ReadME is telling people to use a docker that is not compatible with the examples/notebooks/documentation they then refer them to. Please fix this ASAP.

**Comment by benfred** [comment]
Thanks for the feedback!

It sounds like there are two issues here:

 1) You want containers for a nightly/unstable version of merlin - instead of the stable released version that we have right now.  
 2) We should update the README between releases to indicate this is an unstable/dev version and give links to the stable versions

For the first issue -  we're looking at shipping nightly merlin containers as part of the 0.5 release - I've created an issue here to track https://github.com/NVIDIA/NVTabular/issues/551 .   There is some process involved in shipping to NGC (qa/security scans/perf testing/notifications etc) that we need to fully automate which makes this out of scope for this release

> It seems counterintuitive that the CuDF version is 2 versions behind the currently released one.

We're not two versions behind the currently released version of cudf. The nvtabular 0.4 release will use the most up to date version of cudf when it's released,  just like the nvtabular 0.3 release used the most up to date version of cudf when it was released.

cudf 0.16 was the most recent version when we released nvtabular 0.3 - cudf 0.17 came out a couple weeks after the nvtabular 0.3 release and cudf 0.18 isn't released yet.  Cudf 0.18 is scheduled for [release on Feb 17th](https://docs.rapids.ai/maintainers) - and we will be releasing the next version of NVTabular that uses cudf 0.18 a week after.

> Is there something that breaks when the CuDF version is updated to the current version? If not then why wouldn't you want to update the NGC docker accordingly?

Yes - there have been multiple issues found when upgrading cudf =(  For cudf 0.17 we had https://github.com/NVIDIA/NVTabular/pull/494 (which should be fixed now) and we are still tracking some issues with cudf 0.18 https://github.com/NVIDIA/NVTabular/issues/539 and https://github.com/NVIDIA/NVTabular/issues/538 . These issues should be fixed by the NVTabular 0.4 release

We do stable releases  - including new containers on NGC, releases on github, packages on conda/pypi etc - to provide a stable environment for customers.  I totally agree we should make it easier to get the unstable versions for people to try out, and we should push out nightly containers as part of that though.

**Comment by jperez999** [comment]
Hello @darien-schettler I wanted to take a second and relay that we are working on producing nightly containers that will have the latest releases of all packages including cudf and nvtabular. We have not completed this work yet and it is not live but you can track progress here: https://github.com/NVIDIA/NVTabular/issues/551
